### ISSUE TITLE
[feature] Unsuspending an Account

### ISSUE THREAD

**Issue by zxzzxv** [superseriousbusiness/gotosocial#3539]
### Is your feature request related to a problem ?

While testing out the latest version, I was trying out the reporting system and accidentally suspended a non-test user. The text says account suspensions are irreversible. Are there any plans to make it possible to unsuspend users?

### Describe the solution you'd like.

I'd like something like Mastodon where suspends remove all posts and media, and unsuspends means the server can start to federate with that user as normal.

### Describe alternatives you've considered.

I have considered just nuking my instance and restarting things since it's still mostly a dev instance. No one but test accounts is following the server now. But it would be nice to be able to unsuspend accounts.

**Comment by tsmethurst** [comment]
Good idea yeah, there's a few technical things we'll have to think about there (like do we still send a Delete message out or not), but we can cross that bridge when we come to it.

**Comment by tsmethurst** [comment]
Hummm. Yes, you should be able to undo it by setting `suspended_at` to `NULL` for that account in the `accounts` table.

So something like:

```sql
UPDATE accounts
SET suspended_at = NULL
WHERE username = 'some_username'
AND domain = 'some_domain';
```

You'll then have to restart your GtS instance to invalidate the in-memory cache.

**Comment by daenney** [comment]
We've migrated to Codeberg. Your issue has been imported as https://codeberg.org/superseriousbusiness/gotosocial/issues/3539.
